sky/provision/gcp/instance_utils.py

```diff
@@ -219,6 +237,60 @@ def wait_for_operation(cls, operation: dict, project_id: str,
             return True
         return False
 
+    @classmethod
+    def get_vpc_name(
+        cls,
+        project_id: str,
+        zone: str,
+        instance: str,
+    ) -> Optional[str]:
+        try:
+            response = cls.load_resource().instances().get(
+                project=project_id,
+                zone=zone,
+                instance=instance,
+            ).execute()
+            # Format: projects/PROJECT_ID/global/networks/VPC_NAME
+            vpc_link = response['networkInterfaces'][0]['network']
+            return vpc_link.split('/')[-1]
+        except gcp.http_error_exception() as e:
+            logger.warning(f'Failed to get VPC name for instance {instance}: '
+                           f'{e.reason}. Skip opening ports for it.')
+            return None
+
+    @classmethod
+    def add_tag_if_not_exist(
```

Comment:
Should we rename it to `add_network_tag_if_not_exist` as it is very different from the `tag` in AWS?

Reply:
Done. Thanks!